src/shared/services/auth/auth.js

```diff
@@ -217,146 +31,53 @@ async function handleAuthorizationCallback() {
-
-  storage.removeItem(ACCESS_TOKEN_KEY);
-  storage.removeItem(REFRESH_TOKEN_KEY);
-
-  storage.setItem(STATE_TOKEN_KEY, stateToken);
-  storage.setItem(NONCE_KEY, nonce);
   storage.setItem(OAUTH_DOMAIN_KEY, domain);
-
-  // temporary until Grip is phased out, code should be rewritten and determine flow by oidc.responseType
-  global.location.assign(domain === 'keycloak' ? loginCode(nonce, stateToken) : loginToken(domain, nonce, stateToken));
+  if (domain === 'keycloak') {
+    await keycloak.login();
+  } else {
+    authz.login(domain);
+  }
 }
 
+/**
+ * Performs user logout.
+ */
 export function logout() {
-  storage.removeItem(ACCESS_TOKEN_KEY);
-  storage.removeItem(REFRESH_TOKEN_KEY);
+  const auth = getAuth();
+
   storage.removeItem(OAUTH_DOMAIN_KEY);
+  auth.logout();
 }
 
 /**
- * Initializes the auth service when needed. Catches any callback params and
- * errors from the OAuth2 authorization service when available.
- *
- * When no access token is available it initiates the login process which will
- * redirect the user to the OAuth2 authorization service.
- *
+ * Returns boolean indicating if user is authenticated (with non-expired token).
  */
-export async function initAuth() {
-  restoreAccessToken(); // Restore access token from local storage
-  catchAuthorizationError(); // Catch any error from the OAuth2 authorization service
-  await handleAuthorizationCallback(); // Handle a callback from the OAuth2 authorization service
-
-  if (getOauthDomain() === 'keycloak') {
-    await handleRefreshAccessToken(); // Refresh access token if expired
-    setRefreshAccessTokenInterval(); // Start periodic check for validity of access token, and refresh if expired
-  }
-}
-
-export const isAuthenticated = () => {
-  const decoded = accessTokenParser(getAccessToken());
-
-  if (!decoded?.expiresAt) return false;
-
-  const hasExpired = decoded.expiresAt * 1000 < Date.now();
-
-  return !hasExpired;
-};
-
-export function getScopes() {
-  return tokenData.scopes || [];
-}
-
-export function getName() {
-  return tokenData.name || '';
-}
+export const isAuthenticated = () => getAuth().isAuthenticated();
 
 /**
- * Creates an instance of the native JS `Headers` class containing the
- * authorization headers needed for an API call.
+ * Creates an object containing the authorization headers needed for an API call.
  *
  * @returns {Object<string, string>} The headers needed for an API call.
  */
 export function getAuthHeaders() {
-  const accessToken = getAccessToken();
-  return accessToken ? { Authorization: `Bearer ${accessToken}` } : {};
+  return getAuth().getAuthHeaders();
 }
 
-export const authenticate = async () => {
-  await initAuth();
-
-  const accessToken = getAccessToken();
-
-  if (accessToken) {
-    return { userName: getName(), userScopes: getScopes(), accessToken };
-  }
-
-  return null;
-};
+/**
+ * Perform user authentication on app init.
+ */
+// eslint-disable-next-line require-await
```

Comment:
Could you elaborate?

Reply:
For the authenticate function, the `authz` implementation is synchronous while the `keycloak` implementation is asynchronous (due to how keycloak-js implements the init function). I've added async to this line to have Babel this function promisified. 

Adding `await` which is what the eslint rule suggests, triggers the `no-return-await` linting rule